File: templates/Uwp/Features/DeepLinking.Prism/Activation/SchemeActivationData.cs
using System;
using System.Collections.Generic;
using System.Web;

namespace Param_RootNamespace.Activation
{
    public class SchemeActivationData
    {
        // TODO WTS: Open package.appxmanifest and change the declaration for the scheme (from the default of 'wtsapp') to what you want for your app.
        // More details about this functionality can be found at https://github.com/Microsoft/WindowsTemplateStudio/blob/master/docs/UWP/features/deep-linking.md
        // TODO WTS: Change the image in Assets/Logo.png to one for display if the OS asks the user which app to launch.
        // Also update this protocol name with the same value as package.appxmanifest.
        private const string ProtocolName = "wtsapp";

        public string PageToken { get; private set; }

        public Uri Uri { get; private set; }

        public Dictionary<string, string> Parameters { get; private set; } = new Dictionary<string, string>();

        public bool IsValid => !string.IsNullOrEmpty(PageToken);

        public SchemeActivationData(Uri activationUri)
        {
            PageToken = SchemeActivationConfig.GetPage(activationUri.AbsolutePath);

            if (!IsValid || string.IsNullOrEmpty(activationUri.Query))
            {
                return;
            }

            var uriQuery = HttpUtility.ParseQueryString(activationUri.Query);
            foreach (var paramKey in uriQuery.AllKeys)
            {
                Parameters.Add(paramKey, uriQuery.Get(paramKey));
            }
        }

        public SchemeActivationData(string pageToken, Dictionary<string, string> parameters = null)
        {
            PageToken = pageToken;
            Parameters = parameters;
            Uri = BuildUri();
        }

        private Uri BuildUri()
        {
            var pageKey = SchemeActivationConfig.GetPageKey(PageToken);
            var uriBuilder = new UriBuilder($"{ProtocolName}:{pageKey}");
            var query = HttpUtility.ParseQueryString(string.Empty);

            foreach (var parameter in Parameters)
            {
                query.Set(parameter.Key, parameter.Value);
            }

            uriBuilder.Query = query.ToString();
            return new Uri(uriBuilder.ToString());
        }
    }
}

File: templates/Uwp/Features/DeepLinking._VB/Activation/SchemeActivationData.vb
Imports System.Web

Namespace Activation
    Public Class SchemeActivationData
        ' TODO WTS: Open package.appxmanifest and change the declaration for the scheme (from the default of 'wtsapp') to what you want for your app.
        ' More details about this functionality can be found at https://github.com/Microsoft/WindowsTemplateStudio/blob/master/docs/UWP/features/deep-linking.md
        ' TODO WTS: Change the image in Assets/Logo.png to one for display if the OS asks the user which app to launch.
        ' Also update this protocol name with the same value as package.appxmanifest.
        Private Const ProtocolName As String = "wtsapp"

        Public Property PageType As Type

        Public Property Uri As Uri

        Public Property Parameters As Dictionary(Of String, String) = New Dictionary(Of String, String)()

        Public ReadOnly Property IsValid As Boolean
            Get
                Return PageType IsNot Nothing
            End Get
        End Property

        Public Sub New(activationUri As Uri)
            PageType = SchemeActivationConfig.GetPage(activationUri.AbsolutePath)

            If Not IsValid OrElse String.IsNullOrEmpty(activationUri.Query) Then
                Return
            End If

            Dim uriQuery = HttpUtility.ParseQueryString(activationUri.Query)

            For Each paramKey In uriQuery.AllKeys
                Parameters.Add(paramKey, uriQuery.[Get](paramKey))
            Next
        End Sub

        Public Sub New(pageType As Type, Optional parameters As Dictionary(Of String, String) = Nothing)
            Me.PageType = pageType
            Me.Parameters = parameters
            Me.Uri = BuildUri()
        End Sub

        Private Function BuildUri() As Uri
            Dim pageKey = SchemeActivationConfig.GetPageKey(PageType)
            Dim uriBuilder = New UriBuilder($"{ProtocolName}:{pageKey}")
            Dim query = HttpUtility.ParseQueryString(String.Empty)

            For Each parameter In Parameters
                query.[Set](parameter.Key, parameter.Value)
            Next

            uriBuilder.Query = query.ToString()
            Return New Uri(uriBuilder.ToString())
        End Function
    End Class
End Namespace